pkg/controller/catalog/storage/storage_util.go

```diff
@@ -45,66 +45,47 @@ func (e serviceNotFound) Error() string {
 //
 // If Plan is not given and there's only one plan for a given service, we'll
 // choose that.
-func GetServicePlanInfo(storage Storage, service string, plan string) (string, string, string, error) {
-	s, err := storage.GetServiceClass(service)
+func GetServicePlanInfo(storage ServiceClassStorage, service string, plan string) (string, string, string, error) {
+	s, err := storage.Get(service)
 	if err != nil {
 		return "", "", "", err
 	}
 	// No plan specified and only one plan, use it.
 	if plan == "" && len(s.Plans) == 1 {
-		log.Printf("Found Service Plan GUID as %s for %s : %s", s.Plans[0].ID, service, s.Plans[0].Name)
-		return s.ID, s.Plans[0].ID, s.Plans[0].Name, nil
+		planID := s.Plans[0].CFGUID
+		planName := s.Plans[0].Name
+		log.Printf("Found Service Plan GUID as %s for %s : %s", planID, service, planName)
+		return s.CFGUID, planID, planName, nil
 	}
 	for _, p := range s.Plans {
 		if p.Name == plan {
-			fmt.Printf("Found Service Plan GUID as %s for %s : %s", p.ID, service, plan)
-			return s.ID, p.ID, p.Name, nil
+			planID := p.CFGUID
+			log.Printf("Found Service Plan GUID as %s for %s : %s", planID, service, plan)
+			return s.CFGUID, planID, p.Name, nil
 		}
 	}
 	return "", "", "", servicePlanNotFound{service, plan}
 }
 
 // GetBrokerByServiceClass returns the broker which serves a particular service
 // class.
-func GetBrokerByServiceClass(storage Storage, id string) (*model.ServiceBroker, error) {
+func GetBrokerByServiceClass(
+	brokerStorage BrokerStorage,
+	svcClassStorage ServiceClassStorage,
```

Comment:
I guess we could just pass in a `storage.Storage` here, but that isn't done elsewhere. In all other cases, we pass the _specific_ storage interface that is needed for the task at hand. Here, it just happens that the underlying logic needs to use two such stores.

Side note...

> because we may have 1/2 the system using in-memory and the other 1/2 using etcd

That's not actually true...

In memory storage is currently being used only for testing. etcd-based storage hasn't been implemented yet. The "real" storage impl used at the moment is based on third party resources. (This is a way of cheating and getting k8s to store stuff in its own etcd for us.) That 3pr-based impl likely goes away at some point in the future when we can start shipping the controller with an embedded etcd (a default not suitable for production).

In no case, however, do I believe we're planning on allowing a heterogenous mix of storage impls for different things. You pick one.

So, to bring that answer full circle, that's definitely not a factor in the "split" you see in the various storage interfaces.